src/main/java/com/example/undersstandingmapping/models/Student.java

```diff
@@ -5,9 +5,10 @@
 import jakarta.persistence.OneToMany;
 import lombok.Getter;
 import lombok.Setter;
+import org.springframework.stereotype.Component;
 
 import java.util.List;
-
+@Component
```

Comment:
Why you need to autowire this in controller, please remove @Componet and try again.